### ISSUE TITLE
Added a Perfetto python dump

### ISSUE THREAD

**Issue by alecazam** [GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator#382]
This uses the Catapult flame chart json format for timings.  Allocations are displayed in tracks just like in the png, but with names, and the values can be zoomed in-out and scrolled through.   Perfetto seems to keep trying to kill off this json, without offering an alternative, but it works okay for this.   Timestamp is the offset, and duration correspond to size.   These are all displayed as timings (either ns or ms), but should all be drawn with rect corresponding to the sizes.  Perfetto still doesn't have hover over the values to expand name/duration, but that would help here.

https://github.com/alecazam/kram/blob/main/scripts/GpuMemDumpPerfetto.py

Here's a screenshot of Perfetto with the default dump.  "M" indicates free blocks, since empty string seems to break Perfetto.  And I can't specify colors either.  But unique names each get pseudo-colored in a block track.

<img width="701" alt="image" src="https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/assets/3750359/bf183352-ecdf-4546-8a0e-22d045a42055">

**Comment by alecazam** [comment]
Here's the output from the Sample.json file.  This can be dropped onto the trace viewer at perfetto.dev.

[GpuMemDumpSample.trace.zip](https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/files/13261397/GpuMemDumpSample.trace.zip)

**Comment by alecazam** [comment]
In Perfetto, hit ctrl+shift+P, set_timestamp_format to seconds.  Then the seconds represent MB, and have vertical lines.   Reading the values in seconds provides MB values when blocks are selected.  This works great when all allocations have names, but this Sample.json barely has any.  It's not a real example of a game capture which would have these details present.

**Comment by adam-sawicki-a** [comment]
This is an interesting development. Thank you for sharing this. Although, one could say that using a profiler like this to visualize memory allocations, with allocation sizes faked into time units, is a bit hacky.

**Comment by alecazam** [comment]
It's really no less of a hack than using a static png image to display what should be an interactive display with names of a heap.  There's AMD Memory Visualizer and it curiously skips importing dump files from VMA.  So that would be the next target I would say for a real UI.

**Comment by adam-sawicki-a** [comment]
Yes, I agree, it would be best to develop an interactive GUI tools for browsing VMA JSON dumps.

Radeon Memory Visualizer is a completely different tool with different purpose and has nothing to do with VMA. It captures memory allocation events as made by DX12/Vulkan app on low level, directly from AMD graphics driver.

**Comment by alecazam** [comment]
Sure, but it has many of the same visualizer.  I will say that seeing unnamed blocks in the PNG was useful, but being able to see allocation names and sizes is a game changer.  We used to just draw all this stuff in games using the gpu, and could bring it up in any game.  There are apps like Tracey that do this.
